pyanaconda/modules/localization/localed.py

```diff
@@ -160,12 +161,21 @@ def set_layouts(self, layouts_variants, options=None, convert=False):
 
         layouts = []
         variants = []
+        parsing_failed = False
 
         for layout_variant in (nonempty for nonempty in layouts_variants if nonempty):
-            (layout, variant) = parse_layout_variant(layout_variant)
+            try:
+                (layout, variant) = parse_layout_variant(layout_variant)
+            except InvalidLayoutVariantSpec as e:
+                log.debug("Parsing of %s failed: %s", layout_variant, e)
+                parsing_failed = True
+                continue
             layouts.append(layout)
             variants.append(variant)
 
+        if not layouts and parsing_failed:
+            return
+
         layouts_str = ",".join(layouts)
```

Comment:
What will happen here if parsing will be successful but there won't be any layouts?

Reply:
Hm, I think it that by the logic here, it can't happen.
Or if you mean the case where the method is called with no nonempty layout_variants, then the behaviour should be preserved (which I think is effectively unsetting the X Layouts by supplying empty strings to the SetX11Keyboard). I don't intend to change it in this PR.

Reply:
ok, thanks for explanation.